### PR TITLE
Speed up for classification by refactoring solution interface

### PR DESCRIPTION
**Pull request by jkosata**
`get_single_solution` is supposed to be a user-friendly method but is terribly inefficient for internal work, which has been an issue before (#86 , #130 ) . Rather, we want to pass vectors of variables + varied parameters around, in a fixed order.

Here:
- internal constructions of `OrderedDict` removed (still works as a UI method)
- Jacobians multiple-dispatched to accept both `OrderedDict` and a vector of variables + varied parameters
- classification refactored to directly call `transform_solutions`

`_classify_default` for 1000 solutions of 9 branches, before:

after:

### CODE REVIEW

**Comment by oameye** [comment]
Oeoeo 3x speedup is already very nice! I will review the code today.

File: src/classification.jl [hunk 1]
```diff
@@ -86,12 +69,19 @@ Stable solutions are real and have all Jacobian eigenvalues Re[λ] <= 0.
...
+function is_stable(soln::StateDict, res::Result; kwargs...)
+    _is_stable(values(soln) |> collect, res.jacobian; kwargs...)
 end
 
+function _is_stable(res::Result; kwargs...)
+    _isit(soln) = _is_stable(soln, res.jacobian; kwargs...)
+end
+
+function _is_stable(soln, J; rel_tol=1E-10)
+    λs = eigvals(real.(J(soln)))
```
Comment (oameye) [inline src/classification.jl:81]: This line fails when you classify the solutions for 2d sweeps

**Comment by oameye** [comment]
i don't see the bug. I think it is better you do it.

**Comment by jkosata** [comment]
I'm not seeing this, for example with the `parametron.jl` test, everything runs fine
(there are other issues I'm on though)

**Review by oameye — APPROVED** [review]
I think this is a major improvement! Very nice!

One quick question. What is the idea about batching the threading instead of just using @threads macro of all indices over the solution matrix?

**Comment by oameye** [comment]
@jkosata I will leave merging up to you. If you want you can tag 0.7.3 in the Project.toml and comment `@JuliaRegistrator register` in the commit on the master branch (see [example](https://github.com/NonlinearOscillations/HarmonicBalance.jl/commit/82364caacf19feeee3d8f1184fb797d6153cedaa#commitcomment-134129741)).

**Comment by oameye** [comment]
Before you Tag the master branch to the Julia register, we could also merge the method keyword from branch `method-keyword`.

**Pull request merged**